### PR TITLE
TDD project

### PR DESCRIPTION
**Pull request by george827**
Hello, In this feature branch we,
### **Highlights,**

- [ ] Created a class called `Solver`
- [ ] Created a method called `factorial` that takes one argument, an integer N, and returns the factorial for that number.
- [ ] Created a method called `reverse` that takes one argument, a string word, and returns a word reversed
- [ ] Created a method called `fizzbuzz` that takes one argument, an integer N, and returns a string.
   -   When N is divisible by 3, return "fizz".
   -   When N is divisible by 5, return "buzz".
   -   When N is divisible by 3 and 5, return "fizzbuzz".
   -   Any other case, return N as a string (e.g. say N is 7 then return "7").

### CODE REVIEW

File: spec/reverse_spec.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      expect(Solver.reverse('hello world')).to eq('dlrow olleh')
+    end
+
+    it 'should return "dlrow olleh" when given "hello world"' do
+      expect(Solver.reverse('hello world')).to eq('dlrow olleh')
+    end
+
+    it 'should return "dlrow olleh" when given "hello world"' do
+      expect(Solver.reverse('hello world')).to eq('dlrow olleh')
+    end
```
Comment (Ghiftee) [inline spec/reverse_spec.rb]: - [ ] These test cases are exactly the same. There is no need to repeat it since the same word is passed. Kindly remove the duplicates.

File: spec/fizzbuzz_rspec.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      expect(Solver.fizzbuzz(true)).to eq('Input must be a number')
+    end
+
+    it 'should return an error if input is an array' do
+      expect(Solver.fizzbuzz([1, 2, 3])).to eq('Input must be a number')
+    end
+
+    it 'should return an error if input is a hash' do
+      expect(Solver.fizzbuzz({ a: 1, b: 2, c: 3 })).to eq('Input must be a number')
+    end
```
Comment (Ghiftee) [inline spec/fizzbuzz_rspec.rb]: - [ ] Great work adding tests for all these scenarios, however, you are currently missing a scenario in the requirements: ```Any other case, return N as a string (e.g. say N is 7 then return "7").``` This means that when the number is not divisible by 3 or 5, it should be returned as a string e.g
```expect(Solver.fizzbuzz(7)).to eql('7')  # N is not divisible by 3 or 5```

File: spec/factorials.rb [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    it 'should return 87178291200 when given 14' do
+      expect(Solver.factorial(14)).to eq(87_178_291_200)
+    end
+  end
+
+  # throw an error when input is not a number
+  context 'should throw an error when input is not a number' do
+    it 'should throw an error when given a string' do
+      expect { Solver.factorial('a') }.to raise_error(ArgumentError)
+    end
```
Comment (Ghiftee) [inline spec/factorials.rb]: - [ ] When I run this test, I get this error, kindly fix it

Screenshot 2023-03-17 at 10 30 59

File: spec/factorials.rb [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      expect { Solver.factorial(-1) }.to raise_error(ArgumentError)
+    end
+
+    it 'should throw an error when given a float' do
+      expect { Solver.factorial(1.5) }.to raise_error(ArgumentError)
+    end
+
+    it 'should throw an error when given a nil' do
+      expect { Solver.factorial(nil) }.to raise_error(ArgumentError)
+    end
```
Comment (Ghiftee) [inline spec/factorials.rb]: - [ ] When I run this test, I get this error, kindly fix it

Screenshot 2023-03-17 at 10 31 06

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello @george827 and @tsohleDev  👋🏽 

Good job working on the project so far! 
To highlight 👍🏽 
✔️ Gitflow is used
✔️ TDD is used
✔️ Tests are written for all edge cases

However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by muneebulrehman — APPROVED** [review]
Hi @george827 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🥳
Congratulations! 🎉 

# Status: Approved✔️

## Highlights
- No linter errros
- Descriptive PR
- Good follow-up with the previous reviewer.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
